From 3595a6f3d6c02cd1653836609b0edf098a0927a1 Mon Sep 17 00:00:00 2001
From: rkvenc forward-port <forward-port@local>
Date: Wed, 22 Jul 2026 08:12:29 -0700
Subject: [PATCH 61/87] video: rockchip: rkvdec2-link: test per-core disable
 flag

The CCU work-mode loop tested the outer worker device instead of
the core being iterated. Test core->disable so disabled cores are
not programmed into the work mask.

Signed-off-by: rkvenc forward-port <forward-port@local>
---
 drivers/video/rockchip/mpp/mpp_rkvdec2_link.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/mpp/mpp_rkvdec2_link.c b/drivers/video/rockchip/mpp/mpp_rkvdec2_link.c
index 8ff38b00255fc..a2b03665f504a 100644
--- a/drivers/video/rockchip/mpp/mpp_rkvdec2_link.c
+++ b/drivers/video/rockchip/mpp/mpp_rkvdec2_link.c
@@ -2843,7 +2843,7 @@ static int rkvdec2_hard_ccu_enqueue(struct rkvdec2_ccu *ccu,
 			struct rkvdec2_dev *dec = to_rkvdec2_dev(core);
 			struct rkvdec_link_dev *link_dec = dec->link_dec;
 
-			if (mpp->disable)
+			if (core->disable)
 				continue;
 			work_mode |= dec->core_mask;
 			rkvdec2_ccu_link_fix_rcb_regs(dec);
